pulp_python/app/tasks/sync.py

```diff
@@ -27,6 +29,42 @@
 logger = logging.getLogger(__name__)
 
 
+def update_remote(remote_pk, repo_pks, *args, **kwargs):
```

Comment:
I'm not especially happy with this, but I can see why you've done it this way, and I don't really have a better suggestion.  But we really ought to provide some kind of hook so that it's not necessary to replace all of this code.  Could you file a refactor issue on this?

Reply:
I was thinking about this some more, (I did copy paste this from pulp_ansible), but this doesn't actually need to be here. I can use a similar django_lifecycle hook on the remote and only have the change be in the viewset update method. The only thing that is necessary for the implementation is to have the task gather the locks on the repositories. The default update code will still call the hooks anyway.

Reply:
I believe that will run synchronously with the original update request, rather than as part of the task update.  And actually I think that applies to the hook above as well.

Reply:
Actually I think I know what you mean.  i.e. if one of those fields has changed, add an update to last_serial to the standard update_remote task?

Reply:
What I am saying is that anytime you update an object, it spawns a generic task. Inside that task it get that objects serializer and then call serializer.save(). That save method (unless it is overwritten) will call the model's save method and since every object is a django_lifecycle model, it'll trigger any save/update hooks on that model. Therefor I don't need to create a new task, I can just add a hook and inside the update viewset when it goes to spawn the generic update_task I ensure it has the proper repository locks.

Reply:
+1